File: server/withssafy/src/main/java/com/ssafy/withssafy/service/comment/CommentService.java
package com.ssafy.withssafy.service.comment;

import com.ssafy.withssafy.dto.comment.CommentDto;
import com.ssafy.withssafy.dto.comment.CommentResDto;

import java.util.List;

public interface CommentService {
    List<CommentResDto> findAll();
    List<CommentResDto> findByBoardId(Long BoardId);
    List<CommentResDto> findByUserId(Long UserId);
    CommentResDto insert(CommentDto commentDto);
    CommentResDto update(CommentDto commentDto);
    boolean delete(Long id);
}

File: application/WithSSAFY/app/src/main/java/com/ssafy/withssafy/src/network/service/CommentService.kt
package com.ssafy.withssafy.src.network.service

import com.ssafy.withssafy.src.dto.board.Comment
import com.ssafy.withssafy.util.RetrofitUtil

/**
 * @since 04/27/22
 * @author Jiwoo Choi
 */
class CommentService {

    suspend fun getAllComment() = RetrofitUtil.commentApi.selectCommentAll()

    suspend fun addComment(commentDto: Comment) = RetrofitUtil.commentApi.insertComment(commentDto)

    suspend fun modifyComment(commentDto: Comment) = RetrofitUtil.commentApi.updateComment(commentDto)

    suspend fun deleteComment(commentId: Int) = RetrofitUtil.commentApi.deleteComment(commentId)

    suspend fun getCommentListByBoardId(boardId: Int) = RetrofitUtil.commentApi.selectCommentByBoardId(boardId)

    suspend fun getCommentListByUserId(userId: Int) = RetrofitUtil.commentApi.selectCommentByUserId(userId)
}